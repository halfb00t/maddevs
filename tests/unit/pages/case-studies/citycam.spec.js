import { render } from '@testing-library/vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import CityCam from '@/pages/case-studies/citycam'
import formBaseProps from '../../__mocks__/formBaseProps'

const localVue = createLocalVue()
localVue.use(Vuex)

const DESCRIPTION = 'IoT Case Study: How to monitor traffic in real-time? Read CityCam\'s story of real-time traffic monitoring, which makes transportation safer and faster.'

const META_DATA = {
  description: DESCRIPTION,
  title: 'Mad Devs Case Study: Real-Time Traffic Monitoring System for Cities',
  'og:url': 'https://maddevs.io/case-studies/citycam/',
  'og:type': 'website',
  'og:site_name': 'Custom Software Development Company',
  'og:title': 'Mad Devs Case Study: Real-Time Traffic Monitoring System for Cities',
  'og:description': DESCRIPTION,
  'og:image': 'https://maddevs.io/citycam.png',
  'og:image:width': '1200',
  'og:image:height': '630',
  'twitter:card': 'summary_large_image',
  'twitter:text:title': 'Mad Devs Case Study: Real-Time Traffic Monitoring System for Cities',
  'twitter:description': DESCRIPTION,
  'twitter:image:src': 'https://maddevs.io/citycam.png',
  'twitter:url': 'https://maddevs.io/case-studies/citycam/',
  metaTitle: '',
  url: '',
  jsonLd: '{"@context": "https://schema.org", "@type": "WebPage", "name": "Mad Devs Case Study: Real-Time Traffic Monitoring System for Cities", "description": "IoT Case Study: How to monitor traffic in real-time? Read CityCam\'s story of real-time traffic monitoring, which makes transportation safer and faster.", "publisher": {"@type": "ProfilePage", "name": "Mad Devs Group Limited"}}',
  image: 'https://maddevs.io/blog.png',
}

const mocks = {
  ...formBaseProps,
  $nuxt: {
    $route: {
      push: jest.fn(),
      path: '/citycam',
    },
  },
  $getMediaFromS3: () => 'img.jpg',
  $lazyLoad: {
    init: jest.fn(),
  },
}

const stubs = ['NuxtLink', 'Main']

const directives = {
  'lazy-load': () => {},
}

const actions = {
  setHeaderTransparentArea: () => {},
  setHeaderTransparent: () => {},
}

const getters = {
  homePageContent: () => ({ description: DESCRIPTION }),
}

const store = new Vuex.Store({
  actions,
  getters,
})

describe('CityCam _uid component', () => {
  global.$nuxt = {
    $route: {
      name: null,
    },
  }

  it('should render correctly', () => {
    const { container } = render(CityCam, {
      localVue,
      store,
      mocks,
      stubs,
      directives,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct work head method', () => {
    const wrapper = shallowMount(CityCam, {
      localVue,
      store,
      mocks,
      stubs,
      directives,
    })

    const actual = wrapper.vm.$options.head.call(wrapper.vm)

    actual.meta.forEach(meta => {
      expect(META_DATA[meta.name] || META_DATA[meta.property]).toBe(meta.content)
    })
  })
})
