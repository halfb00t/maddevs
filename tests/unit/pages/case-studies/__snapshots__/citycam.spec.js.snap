// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityCam _uid component should render correctly 1`] = `
<div>
  <main
    class="case case--citycam"
  >
    <section
      class="case_header case_header-citycam"
      id="transparent-header-area"
    >
      <img
        alt="case video fallback"
        class="case_main-video_fallback"
        data-src="img.jpg"
      />
       
      <!---->
       
      <div
        class="case_header-content"
      >
        <div
          class="case_header-text"
          style="opacity: 1;"
        >
          <div
            class="case_case-study-item"
          >
            
        Case Study
      
          </div>
           
          <h1
            class="case_header-title"
          >
            
      Better cities 
            <br />
             with CityCam
    
          </h1>
           
          <p
            class="case_header-description"
          >
            
      Building hardware and software for energy-efficient 
            <br />
             and highly autonomous traffic cameras
    
          </p>
           
        </div>
         
        <img
          alt="CityCam intersection monitoring system"
          class="case_header-logo case_citycam-logo"
          height="93"
          src="img.jpg"
          style="opacity: 1;"
          width="253"
        />
      </div>
    </section>
     
    <main-stub />
     
    <section
      class="case_footer container_full"
    >
      <img
        alt="Peklo Tool for search ad campaigns"
        class="case_logotype-peklo"
        data-src="[object Object]"
        height="72"
        width="195"
      />
       
      <h2
        class="case_title case_title_h2 case_title-peklo"
      >
        
    Peklo Tool for better
    ad campaigns
  
      </h2>
       
      <nuxtlink-stub
        class="case_next-project-button"
        to="/case-studies/peklo/"
      >
        
    Read next project
  
      </nuxtlink-stub>
    </section>
  </main>
</div>
`;
